### ISSUE TITLE
Create folder via api not possible "No destination folder provided"

### ISSUE THREAD

**Issue by michast** [nilsteampassnet/TeamPass#2335]
### Steps to reproduce
1. Enable API
2. Create key
3. curl "http://teampass.my.domain.de/api/index.php/add/folder/myfolder;25;16;0;0?apikey=thisismyawesomekey" from bash or in Firefox

### Expected behaviour
Create the folder "myfolder" below the folder with the ID 16.

### Actual behaviour
Message in browser / curl: ' err | "No destination folder provided" '
Message in httpd-log: PHP Notice:  Undefined offset: 1 in /data/teampass/api/functions.php

After some debugging I found out that the function "Urlsafe_b64decode" causes the message. When I transfer the data without being processed by the function, everything works.
Change this:
$params = explode(';', Urlsafe_b64decode($GLOBALS['request'][2]));
to
$params = explode(';', $GLOBALS['request'][2]);

### Server configuration
**Operating system**:
RHEL 7

**Web server:**
Server version: Apache/2.4.6 (Red Hat Enterprise Linux)
Server built:   Jan  8 2018 06:49:04

**Database:**
mysql  Ver 15.1 Distrib 5.5.56-MariaDB, for Linux (x86_64) using readline 5.1

**PHP version:**
PHP 7.2.7 (cli) (built: Jun 20 2018 08:21:26) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.2.0, Copyright (c) 1998-2018 Zend Technologies

**Teampass version:** 
2.1.27.16

**Teampass configuration file:**

**Updated from an older Teampass or fresh install:**
Fresh install

### Client configuration
**Browser:**
- curl 7.47.0 (x86_64-pc-linux-gnu) libcurl/7.47.0 GnuTLS/3.4.10 zlib/1.2.8 libidn/1.32 librtmp/2.3
- Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:60.0) Gecko/20100101 Firefox/60.0

**Operating system:**
Linux Mint 18.3 Sylvia

**Comment by nilsteampassnet** [comment]
Please provide your api version.

> `https://teampass.readthedocs.io/en/latest/api/api-special/#api-version`

I believe its quite old because this part of the code isn't anymore and has been improved.

I'm closing the ticket but you can keep exchange

**Comment by michast** [comment]
I took a closer look at it after installing an upgrade.
I had installed version 2.1.27.16, which is probably the most recent. I had this version again installed as an upgrade. The API remains at 2.1.1.

After that I looked around the different branches.
In Branch 2.1.27 [()](https://github.com/nilsteampassnet/TeamPass/blob/2.1.27/api/functions.php) the API seems to be correct (version 2.1.26).

The master branch [()](https://github.com/nilsteampassnet/TeamPass/blob/master/api/functions.php) is version 2.1.1.

@nilsteampassnet 
Is this possibly the fault?
